include/swift/AST/SubstitutionMap.h

```diff
@@ -99,6 +97,9 @@ class SubstitutionMap {
   SubstitutionMap subst(TypeSubstitutionFn subs,
                         LookupConformanceFn conformances) const;
 
+  /// Build an array of substitutions from a substitution map.
+  void toList(SmallVectorImpl<Substitution> &subs) const;
+
```

Comment:
I think you guess it, but an example of compatible generic signatures could be e.g. `<T, U>` and `<T, U where U == Int>`. Both have the same set of generic type parameters in the same order, but the second generic signature requires only one substitution.